### ISSUE TITLE
Debugging Groovy incorrect

### ISSUE THREAD

**Issue by blakemcbride** [apache/netbeans#4715]
### Apache NetBeans version

Apache NetBeans 15

### What happened

I am debugging a Groovy program.  Once a breakpoint is hit, I am using "Step Over" to step through the lines.  The debugger is going to lines that it should not.  

For example, I hit a breakpoint at the following point:

![Pic1](https://user-images.githubusercontent.com/6467584/193430149-8451d37e-4091-47a0-bbfb-7a45a9fd5880.png)

After hitting "Step Over" NetBeans goes to the next line as expected:

![Pic2](https://user-images.githubusercontent.com/6467584/193430173-6b4d921d-bf18-43c8-a960-83c967de8108.png)

Again, "Step Over" is hit, and NetBeans goes to the correct line:

![Pic3](https://user-images.githubusercontent.com/6467584/193430201-113d0370-5871-4f57-8a3c-4062aa3cd7e3.png)

Once again, I get the "Step Over" button, and NetBeans goes to the wrong line:

![Pic4](https://user-images.githubusercontent.com/6467584/193430218-70db3217-6328-461d-b423-0cbd81a9d8e6.png)

Interestingly, rather than execute the line it is on (a return) it skips over it when I hit the "Step Over" button:

![Pic5](https://user-images.githubusercontent.com/6467584/193430238-7b6d09a5-20bc-4171-b72d-9c2996bb91f3.png)

So, basically, it showed that it stopped at the return statement when the return statement should not (and did not) occur at all.

Thanks!

### Did this work correctly in an earlier version?

No / Don't know

### Operating System

Fedora Linux

### JDK

17

### Apache NetBeans packaging

Apache NetBeans binary zip

### Anything else

I took extra care to be sure there wasn't a disconnect between the source code and the compiled version.  In other words, they're not out of sync.

This is just one small example that I took the time to document.  The Groovy debugger has many problems.

### Are you willing to submit a pull request?

No

### Code of Conduct

Yes

**Comment by blakemcbride** [comment]
This is better described at [https://github.com/apache/netbeans/issues/4754](https://github.com/apache/netbeans/issues/4754)
